src/models/reposyncstate.ts

```diff
@@ -0,0 +1,127 @@
+import Sequelize, { Op } from "sequelize";
+import Subscription, { RepositoryData, RepoSyncStateObject, TaskStatus } from "./subscription";
+
+export default class RepoSyncState extends Sequelize.Model {
+	id: number;
+	subscriptionId: number;
+	repoId: number;
+	repoName: string;
+	repoOwner: string;
+	repoFullName: string;
+	repoUrl: string;
+	priority?: number;
+	branchStatus?: TaskStatus;
+	commitStatus?: TaskStatus;
+	issueStatus?: TaskStatus;
+	pullStatus?: TaskStatus;
+	buildStatus?: TaskStatus;
+	deploymentStatus?: TaskStatus;
+	branchCursor?: string;
+	commitCursor?: string;
+	issueCursor?: string;
+	pullCursor?: string;
+	buildCursor?: string;
+	deploymentCursor?: string;
+	forked?: boolean;
+	repoPushedAt?: Date;
+	repoUpdatedAt?: Date;
+	repoCreatedAt?: Date;
+	syncUpdatedAt?: Date;
+	syncCompletedAt?: Date;
+	updatedAt: Date;
+	createdAt: Date;
+
+	static async findAllFromSubscription(subscription: Subscription): Promise<RepoSyncState[]> {
+		return RepoSyncState.findAll({
+			where: {
+				subscriptionId: subscription.id
+			}
+		});
+	}
+
+	static async deleteAllFromSubscription(subscription: Subscription): Promise<number> {
+		return RepoSyncState.destroy({
+			where: {
+				subscriptionId: subscription.id
+			}
+		});
+	}
+
+	static async updateFromJson(subscription: Subscription, json: RepoSyncStateObject = {}): Promise<RepoSyncState[]> {
```

Comment:
nit: I feel like `json` might be not the best name here, since technically we are passing an object, not a json string...

Reply:
I was thinking more of db 'JSON' type